rpcs3/Emu/Memory/vm.cpp

```diff
@@ -512,9 +512,9 @@ namespace vm
 				to_clear = for_all_range_locks(to_clear, [&](u32 addr2, u32 size2)
 				{
 					// TODO (currently not possible): handle 2 64K pages (inverse range), or more pages
-					if (g_shareable[addr2 >> 16]) [[unlikely]]
+					if (u64 is_shared = g_shmem[addr2 >> 16]) [[unlikely]]
 					{
-						addr2 &= 0xffff;
+						addr2 = static_cast<u16>(addr2) | is_shared;
```

Comment:
You lose upper 32 bits here.

Reply:
Oh right, address_range is completely for 32-bit address space...